tests/neg/boundspropagation.scala

```diff
@@ -25,19 +25,6 @@ object test3 {
   }
 }
 
-// Example contributed by Jason.
```

Comment:
This test has been removed as we discussed with @AleksanderBG and we think it is correct for it to actually compile.

My reasoning is following:
we have `y: Tree[S, T]` where `Nothing <: S`, `Option[Nothing] <: Option[S] <: T`
we ascribe y: Tree[Nothing, Option[Nothing]] 
 - we can do that because:
    `Nothing <: S`, `Option[Nothing] <: T` so `Tree[S, T] <: Tree[Nothing, Option[Nothing]]`
    (because `Tree` is contravariant in both its parameters)

 we want y to conform to `Tree[A,B]` where `B <: Option[N]`, as `Option[Nothing] <: Option[N]`, it typechecks

Reply:
Shouldn't this be re added as a pos test?

Reply:
Indeed, I have added it.